src/vm/ast/definition_sorter/mod.rs

```diff
@@ -53,6 +54,14 @@ impl<'a> DefinitionSorter {
         Ok(())
     }
 
+    pub fn run_pass_free(
```

Comment:
Can `run_pass_free` be implemented by passing a "free" LimitedCostTracker to `pass.run()` instead of creating the new `run_free()` function?